Assets/MRTK/Core/Utilities/StandardShader/ClippingPrimitive.cs

```diff
@@ -206,11 +239,37 @@ public void ClearRenderers(bool autoDestroyMaterial = true)
         /// <returns>The current list of renderers.</returns>
         public IEnumerable<Renderer> GetRenderersCopy()
         {
-            return new List<Renderer>(renderers);
+            return renderers.AsReadOnly();
         }
 
         #region MonoBehaviour Implementation
 
+        protected void Awake()
+        {
+            if (renderers.Count != rendererDict.Count)
+            {
+                rendererDict.Clear();
+                foreach (var renderer in renderers)
+                {
+                    if (applyToSharedMaterial)
+                    {
+                        rendererDict[renderer] = null;
+                    }
+                    else
+                    {
+                        MaterialInstance materialInstance = renderer.EnsureComponent<MaterialInstance>();
+                        rendererDict[renderer] = materialInstance;
+                        materialInstance.CacheSharedMaterialsFromRenderer = CacheSharedMaterialsFromRenderer;
+                    }
+                }
+            }
+            if (renderers.Count != rendererDict.Count)
```

Comment:
Is this case ever possible to hit? It feels like it should be accounted for in the above `if`, where we populate the dictionary directly from the list.

Reply:
I put it here in case there are duplicates in `renderers`. It is theoretically possible as `renderers` is a list. I don't really expect this to happen when people know what they are supposed to do but still put it here to make sure the list and the dict are always in sync after `Awake()`.